src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -235,4 +238,47 @@ public boolean equals(Object obj) {
                 && filteredFacilitators.equals(other.filteredFacilitators);
     }
 
+
+    //=========== Task ================================================================================
+
+    @Override
+    public boolean hasTask(Task task) {
+        requireNonNull(task);
+        return addressBook.hasTask(task);
+    }
+
+    @Override
+    public void deleteTask(Task task) {
+        addressBook.removeTask(task);
+    }
+
+    @Override
+    public void addTask(Task task) {
+        addressBook.addTask(task);
+        updateFilteredTaskList(PREDICATE_SHOW_ALL_TASKS);
+    }
+
+    @Override
+    public void setTask(Task target, Task editedModule) {
```

Comment:
I think can consider changing editedModule to editedTask

Reply:
Resolved. Thanks!